Index: nautilus-2.26.0/nautilus.desktop.in
===================================================================
--- nautilus-2.26.0.orig/nautilus.desktop.in	2009-03-16 12:49:09.000000000 +0100
+++ nautilus-2.26.0/nautilus.desktop.in	2009-04-01 11:21:32.000000000 +0200
@@ -14,5 +14,5 @@
 X-GNOME-Bugzilla-Version=2.26.0
 X-GNOME-Autostart-Phase=Desktop
 X-GNOME-Autostart-Notify=true
-X-GNOME-AutoRestart=true
+X-GNOME-AutoRestart=false
 X-GNOME-Provides=filemanager
Index: nautilus-2.26.0/nautilus.desktop
===================================================================
--- nautilus-2.26.0.orig/nautilus.desktop	2009-03-16 12:51:15.000000000 +0100
+++ nautilus-2.26.0/nautilus.desktop	2009-04-01 11:21:32.000000000 +0200
@@ -110,5 +110,5 @@
 X-GNOME-Bugzilla-Version=2.26.0
 X-GNOME-Autostart-Phase=Desktop
 X-GNOME-Autostart-Notify=true
-X-GNOME-AutoRestart=true
+X-GNOME-AutoRestart=false
 X-GNOME-Provides=filemanager
Index: nautilus-2.26.0/nautilus.desktop.in.in
===================================================================
--- nautilus-2.26.0.orig/nautilus.desktop.in.in	2009-04-01 11:21:36.000000000 +0200
+++ nautilus-2.26.0/nautilus.desktop.in.in	2009-04-01 11:22:00.000000000 +0200
@@ -14,5 +14,5 @@
 X-GNOME-Bugzilla-Version=@VERSION@
 X-GNOME-Autostart-Phase=Desktop
 X-GNOME-Autostart-Notify=true
-X-GNOME-AutoRestart=true
+X-GNOME-AutoRestart=false
 X-GNOME-Provides=filemanager
