#
# Description: translate the mount-archive menu entry
#
Index: nautilus-2.25.2/po/POTFILES.in
===================================================================
--- nautilus-2.25.2.orig/po/POTFILES.in	2008-12-15 16:38:07.000000000 +0100
+++ nautilus-2.25.2/po/POTFILES.in	2008-12-17 14:27:34.000000000 +0100
@@ -5,6 +5,7 @@
 cut-n-paste-code/libegg/eggsmclient.c
 data/browser.xml
 data/nautilus.xml.in
+debian/mount-archive.desktop.in
 eel/eel-alert-dialog.c
 eel/eel-canvas.c
 eel/eel-editable-label.c
